addons/l10n_de/migrations/2.0/pre-migrate.py

```diff
@@ -12,6 +12,36 @@ def rename_tag(cr, old_tag, new_tag):
 
 
 def migrate(cr, version):
+    # By deleting tag B from ir_model_data we ensure that the ORM won't try to remove this record.
+    # This is done because the tag might be already used as a FK somewhere else.
+    cr.execute(
+        """DELETE FROM ir_model_data
+           WHERE module='l10n_de'
+           AND name='tag_de_liabilities_bs_B'
+        """
+    )
+
+    # As some people already upgraded, they will have renamed the C_1 tag to B_1. This doesn't come from the script but from the
+    # account_account_tags_data.xml. If they try to run this script now in the fix they get the error that B_1 already exists.
+    # To fix this we can check if it exists or not, and if it does then we don't run the script. This means that the ones
+    # that upgraded won't have the old tags data transferred to the new tags but they will still be able to have the updated sheet.
+    cr.execute(
+        """SELECT 1 FROM ir_model_data
+           WHERE module='l10n_de' AND name='tag_de_liabilities_bs_B_1'
+        """)
+    if cr.rowcount:
+        # If the script didn't run, we should remove the tags that have been replaced from ir_model_data too so they're
+        # not deleted by the ORM if they were already used. We wouldn't need to do this if the script ran and tag F
+        # was renamed to tag E.
```

Comment:
It gives an example of why we do this, that tag F would've been renamed to tag E but since the script didn't run, F was not renamed, it still exists in the database but not in the tags data xml, so the ORM tries to remove it which ends up giving errors if the tag was already used in another table as a FK.
Do you think I should explain more or just remove that line?

Reply:
I will say remove it.